all: fix various typos across the whole repo

koibtw opened the pull request:
i used https://github.com/crate-ci/typos to automatically spellcheck and apply fixes to the codebase.
every single line is manually reviewed by me. no breaking changes.
i also added a workflow to automatically find typos :)

please carefully review the changes in case i missed something.

mertcandav commented on std/math/big/wordprime.jule:
```diff
@@ -136,7 +136,7 @@ fn probablyPrimeLucasW(mut n: []Word): bool {
 	}
 
-	// Grantham definition of "extra strong Lucas pseudoprime", after Thm 2.3 on p. 876
+	// Grantham definition of "extra strong Lucas pseudoprime", after Them 2.3 on p. 876
```
```suggestion
	// Grantham definition of "extra strong Lucas pseudoprime", after Thm 2.3 on p. 876
```
This is not a typo. It's stand for `Theorem`.

koibtw replied:
oops, must've missed this one

koibtw committed:
fix: various typos across the whole repo

i used https://github.com/crate-ci/typos to automatically spellcheck and apply fixes to the codebase.
every single line is manually reviewed by me. no breaking changes.
i also added a workflow to automatically find typos :)

Signed-off-by: Adam Perkowski <adas1per@protonmail.com>

mertcandav reviewed (approved):
Yes. Problem is caused by the master and latest IR.
Your changes looks good to me. Thanks for your contribution.
